docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+
+## Mig Cluster
+
+Local Mig cluster: 
+```
+ 
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-local
+  namespace: openshift-migration
+spec:
+  # [!] Change isHostCluster to 'false' if you want to use a clusterRef and serviceAccountSecretRef
+  #     instead of using the mig-controller detected kubeconfig. Refer to mig-cluster-aws.yaml for an example.
+  isHostCluster: true
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+
+  Remote Mig cluster:
+
+  ```
+  apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-remote
+  namespace: openshift-config
+spec:
+  # [!] Change isHostCluster to 'true' if you'll be running the controller on this cluster.
+  #     Setting 'isHostCluster' to true will bypass using the clusterRef and serviceAccountSecretRef below.
+  isHostCluster: false
+
+  url: "https://my-remote-cluster-ip.nip.io:8443"
+
+  serviceAccountSecretRef:
+    name: sa-token-remote
+    namespace: openshift-config
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+- spec.azureResourceGroup: For azure clusters -- it's the resource group that in-cluster volumes use.
+
+- spec.caBundle: If the migcluster needs SSL verification for connections a user can supply a custom CA bundle. This field is required only when spec.Insecure is set false.
+  
+- spec.exposedRegistryPath: Stores the path of registry route when using direct migration.
+
+- spec.insecure: If set false, user will need to provide CA bundle for TLS connection to the remote cluster.
+
+- spec.isHostCluster: Specifies if the cluster is host (where the controller is installed) or not. This is a required field.
+
+- spec.refresh: If set True, forces the controller to run a full suite of validations on migcluster.
+
+- spec.restartRestic: An override setting to tell the controller that the source cluster restic needs to be restarted after stage pod creation.
+
+- spec.serviceAccountSecretRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.url: Stores the url of the remote cluster. The field is only required for the source cluster object.
+
+## MigHooks
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigHook
+metadata:
+  generateName: test-
+  name: test-btmv5
+  namespace: openshift-migration
+spec:
+  custom: false
+  image: 'quay.io/konveyor/hook-runner:latest'
+  playbook: >-
+    LSBuYW1lOiBTbGVlcCBmb3IgMTAgc2Vjb25kcyBhbmQgY29udGludWUgd2l0aCBwbGF5CiAgd2FpdF9mb3I6CiAgIadsfwerq0OiAxMA==
+  targetCluster: source
+```
+
+- spec.activeDeadlineSeconds: Specifies the highest amount of time for which the hook will run.
+
+- spec.custom: Specifies whether the hook is a custom Ansible playbook or a pre-built image. This is a required field.
```

Comment:
@eriknelson @shubham-pampattiwar 

spec.custom: Does `true` mean that the hook is a playbook or an image? This is not clear to me.

Reply:
@apinnick `true` means the hooks is an image.

Reply:
@apinnick @JaydipGabani If hook.custom is true they are saying they are using a custom image. If they are using a custom image they can still use a playbook with it. They could do this in order to use an image with additional ansible modules our image doesn't provide, for example.The description there isn't very good.